src/libs/ajax/DatasetBuilder.ts

```diff
@@ -1,11 +1,93 @@
+// Types that can be used to create a criteria.
+
+export interface DomainType {
```

Comment:
I think most of these types can loose the Type postfix... which I think is only really needed for enum-like types like DataType below.   A more useful name for this specific type could be XxxxDomain to be more descriptive.... maybe DatasetDomain?

Reply:
The reason why they have `Type` is not because they're types in typescript, but because they describe the type (class) of objects in the dataset that can be used to create a criteria. For example, a ProgramDataListType may represent Gender and can be used to create a Criteria that filters the cohort on Gender.

So you have DomainType -> DomainCriteria (actually it's an array of DomainCriteria but we're not modeling that yet), ProgramDataListType -> ProgramDataListCriteria, and ProgramDataRangeType -> ProgramDataRangeCriteria.

With Domain I can see calling it something else, like DatasetDomain, but I don't see a similar name to describe all the list or range based types you can use to generate a list or range criteria. Do you have a better name in mind for ProgramDataListType and ProgramDataRangeType?

Reply:
Hmmm... maybe ProgramDataListFilter and ProgramDataRangeFilter?  (suggested the same in comment reply elsewere)

Reply:
See conversation below https://github.com/DataBiosphere/terra-ui/pull/3939/files/52ca1ceb581de85b211b5d123ba52e3bd945966c#r1218570190